openpgp/keys.go

```diff
@@ -361,7 +361,14 @@ EachPacket:
 
 			// First handle the case of: a self-signature, and no previous self-signature
 			// verified for this UID.
-			if current != nil && current.SelfSignature == nil && (pkt.SigType == packet.SigTypePositiveCert || pkt.SigType == packet.SigTypeGenericCert) && pkt.IssuerKeyId != nil && *pkt.IssuerKeyId == e.PrimaryKey.KeyId {
+			// According to RFC8440, Section 5.2.3.3, if there are several self-signatures,
+			// we should take the newer one.
+			if current != nil &&
+				(current.SelfSignature == nil || pkt.CreationTime.After(current.SelfSignature.CreationTime)) &&
+				(pkt.SigType == packet.SigTypePositiveCert || pkt.SigType == packet.SigTypeGenericCert) &&
+				pkt.IssuerKeyId != nil &&
+				*pkt.IssuerKeyId == e.PrimaryKey.KeyId {
```

Comment:
Should we separate this out into a separate function? I think it's important to highlight that

    current.SelfSignature == nil || pkt.CreationTime.After(current.SelfSignature.CreationTime)

is the really important part that implements the comment above. But the monolithic expression makes it hard to do that.